test/flows/plan_adoption_leave_flow_test.rb

```diff
@@ -0,0 +1,77 @@
+require "test_helper"
+require "support/flow_test_helper"
+
+class PlanAdoptionLeaveFlowTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup { testing_flow PlanAdoptionLeaveFlow }
+
+  should "render a start page" do
+    assert_rendered_start_page
+  end
+
+  context "question: child_match_date?" do
+    setup { testing_node :child_match_date? }
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of child_arrival_date? for a response of 3 months ago" do
+        assert_next_node :child_arrival_date?, for_response: 3.months.ago.to_s
+      end
+    end
+  end
+
+  context "question: child_arrival_date?" do
+    setup do
+      testing_node :child_arrival_date?
+      add_responses child_match_date?: 3.months.ago.to_s
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of leave_start? for a response that is after the child_match_date" do
+        assert_next_node :leave_start?, for_response: 1.day.after(3.months.ago).to_s
+      end
+
+      context "validation" do
```

Comment:
this context is conventionally before `context "next_node"` as per: https://github.com/alphagov/smart-answers/blob/8d4d04d61c3ede76358fb10cf2470e94d4fa68a1/test/flows/state_pension_age_flow_test.rb#L35-L47